docs/WAIVERS.md

```diff
@@ -137,17 +108,87 @@ returning `False` if the RPM is not installed.
     rhel == 8 and 'some thing' in note and env('INFRA') == 'jenkins'
 ```
 
-### Expecting both `pass` and `fail`/`error`
+## Collecting a list of sections
 
-Sometimes, failures or errors are not reliable and happen only occassionally.
-This is why waivers are "permissive" by default - they only waive
-`fail`/`error`, but allow `pass` to pass through, only noting `waived pass`
-in the result note.
+Sections (as defined above) are gathered from multiple waiver files
+in alphanumerical order, and - within each file - from the top.
 
-To override this, pass `strict=True` to `Match()`, which tells the waiving code
-to trigger a waive failure when the waiver matches `pass`.
+This creates a one big unified sequential ordered list of sections
+that is later used for waiving.
 
-```python
-/some/result/name
-    Match(rhel >= 8, strict=True)
+## How a result is waived
+
+When `results.report()` is called by test code to report a new result,
+the waiving logic looks at the status (`pass`, `fail`, etc.).
+
+If the status is `info` or `warn`, no further processing is done, and
+the result remains intact.  
+All other statuses continue below.
+
+The waiving logic then goes through the big list of sections (gathered
+above), **from beginning to end**, and tries to match the test name
+contained within the result, against all regexps in each section.
+
+For example, a test name of `/some/result/name`, when evaluated against
+```
+/different/name.*
+    True
+
+/some/unrelated/string
+/some/.+/name
+    True
```

Comment:
Switch the regexes and use more interesting condition than just `True`

```suggestion
/some/.+/name
/some/unrelated/string
    rhel >= 8
```

It will be useful for further explanation

Reply:
Agree with the non-True condition, but I wanted to emphasize that it's not the first regex that has to match - it can be any of them. And for that, putting it second seemed better.

Reply:
And I'd like to show it doesn't have to be right next to the regex :D What about putting 3 regexes with 2nd as the match?

Reply:
Sure.